src/RadhydroPulseMGint/test_radhydro_pulse_MG_int.cpp

```diff
@@ -158,31 +165,53 @@ auto compute_kappa(const double nu, const double Tgas) -> double
 	// cm^-1
 	auto T_ = Tgas / T_ref;
 	auto nu_ = nu / nu_ref;
-	return kappa0 * std::pow(T_, -0.5) * std::pow(nu_, -3) * (1.0 - std::exp(-coeff_ * nu_ / T_));
+	return scaleup * kappa0 * std::pow(T_, -0.5) * std::pow(nu_, -3) * (1.0 - std::exp(-coeff_ * nu_ / T_));
 }
 
 template <>
 AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE auto
 RadSystem<MGProblem>::DefineOpacityExponentsAndLowerValues(amrex::GpuArray<double, nGroups_ + 1> const rad_boundaries, const double rho,
-							   const double Tgas) -> amrex::GpuArray<amrex::GpuArray<double, nGroups_>, 2>
+							   const double Tgas) -> amrex::GpuArray<amrex::GpuArray<double, nGroups_ + 1>, 2>
 {
-	amrex::GpuArray<double, nGroups_> exponents{};
-	amrex::GpuArray<double, nGroups_> kappa_lower{};
-	for (int g = 0; g < nGroups_; ++g) {
-		auto kappa_up = compute_kappa(rad_boundaries[g + 1], Tgas);
-		auto kappa_down = compute_kappa(rad_boundaries[g], Tgas);
-		exponents[g] = std::log(kappa_up / kappa_down) / std::log(rad_boundaries[g + 1] / rad_boundaries[g]);
-		kappa_lower[g] = kappa_down / rho;
+	amrex::GpuArray<double, nGroups_ + 1> exponents{};
+	amrex::GpuArray<double, nGroups_ + 1> kappa_lower{};
+
+	for (int g = 0; g < nGroups_ + 1; ++g) {
+		if (RadSystem_Traits<MGProblem>::opacity_model == OpacityModel::piecewise_constant_opacity) {
+			exponents[g] = 0.0;
```

Comment:
Can this be upgraded to an if constexpr?

Reply:
Yes. Done. Originally I removed constexpr because I used `out-of-line-definition` rad_boundaries_. Now I changed it to passing rad_boundaries as an argument and there is no problem anymore.